library/VTKExtensions/Readers/vtkF3DEXRReader.cxx

```diff
@@ -0,0 +1,146 @@
+#include "vtkF3DEXRReader.h"
+
+#include "vtkFloatArray.h"
+#include "vtkImageData.h"
+#include "vtkNew.h"
+#include "vtkObjectFactory.h"
+#include "vtkPointData.h"
+#include "vtksys/FStream.hxx"
+
+#include <ImfArray.h>
+#include <ImfRgbaFile.h>
+
+#include <sstream>
+
+vtkStandardNewMacro(vtkF3DEXRReader);
+
+//------------------------------------------------------------------------------
+vtkF3DEXRReader::vtkF3DEXRReader() = default;
+
+//------------------------------------------------------------------------------
+vtkF3DEXRReader::~vtkF3DEXRReader() = default;
+
+//------------------------------------------------------------------------------
+void vtkF3DEXRReader::PrintSelf(ostream& os, vtkIndent indent)
+{
+  this->Superclass::PrintSelf(os, indent);
+}
+
+//------------------------------------------------------------------------------
+void vtkF3DEXRReader::ExecuteInformation()
+{
+  // Setup filename to read the header
+  this->ComputeInternalFileName(this->DataExtent[4]);
+  if (this->InternalFileName == nullptr || this->InternalFileName[0] == '\0')
+  {
+    return;
+  }
+
+  try
+  {
+    Imf::RgbaInputFile file(this->InternalFileName);
+
+    Imath::Box2i dw = file.dataWindow();
+    this->DataExtent[0] = dw.min.x;
+    this->DataExtent[1] = dw.max.x;
+    this->DataExtent[2] = dw.min.y;
+    this->DataExtent[3] = dw.max.y;
+
+    Imf::RgbaChannels channels = file.channels();
+    if (channels != Imf::RgbaChannels::WRITE_RGBA && channels != Imf::RgbaChannels::WRITE_RGB)
+    {
+      throw std::runtime_error("type of channels not supported yet");
```

Comment:
what type of channel do we not support ?

Reply:
There's a bunch of different formats in OpenEXR but I've only seen RGB or RGBA files for now.